infra/terraform/modules/data_access/memberships/memberships.py

```diff
@@ -52,10 +53,35 @@ def attach_bucket_policy(event, context):
     team_slug = event["team"]["slug"]
 
     client = boto3.client("iam")
-    client.attach_role_policy(
-        RoleName=naming.role_name(username),
-        PolicyArn=policy_arn(team_slug, policy_type),
-    )
+
+    def fn():
+        client.attach_role_policy(
+            RoleName=naming.role_name(username),
+            PolicyArn=policy_arn(team_slug, policy_type),
+        )
+
+    retry(fn)
+
+
+def retry(fn, attempts=5, delay=0.150):
+    tries = attempts
+    while tries > 0:
+        try:
+            fn()
+            break
+        except botocore.exceptions.ClientError as error:
+            if error.response["Error"]["Code"] != "NoSuchEntity":
+                # Only retry on boto's ClientError/NoSuchEntity error
+                raise
+
+            tries -= 1
+            LOG.warning(
+                "error while attaching role to policy: {}.".format(error))
+            if tries > 0:
+                LOG.warning("Retry in {}s...".format(delay))
+                time.sleep(delay)
```

Comment:
If you `continue` here, you can remove the `break` from inside the `try` clause and, combined with inverting the logic of the error code check, you can remove one of the `raise` statements by just falling through both `if` statements, eg:
```python
while tries > 0:
    try:
        fn()
    except ... as error:
        if error... == 'NoSuchEntity':
            tries -= 1
            if tries > 0:
                time.sleep(delay)
                continue
        raise
```